src/states/features/auth/authSlice.js

```diff
@@ -6,6 +6,7 @@ const role = false;
 
 const initialState = {
   token: currentToken,
+  email: false,
```

Comment:
Instead of using separate states, you can retrieve the payload from the token itself. This way, when a user logs in, you can extract the token, decode it, and access the data, including the email. By directly accessing the token payload, you can obtain the necessary information without the need for additional states or variables